google/resource_storage_bucket_acl.go

```diff
@@ -138,53 +134,37 @@ func resourceStorageBucketAclCreate(d *schema.ResourceData, meta interface{}) er
 			return fmt.Errorf("Error updating bucket %s: %v", bucket, err)
 		}
 
-		return resourceStorageBucketAclRead(d, meta)
 	}
 
-	return nil
+	d.SetId(getBucketAclId(bucket))
+	return resourceStorageBucketAclRead(d, meta)
 }
 
 func resourceStorageBucketAclRead(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
 
 	bucket := d.Get("bucket").(string)
 
-	// Predefined ACLs cannot easily be parsed once they have been processed
-	// by the GCP server
-	if _, ok := d.GetOk("predefined_acl"); !ok {
-		role_entity := make([]interface{}, 0)
-		re_local := d.Get("role_entity").([]interface{})
-		re_local_map := make(map[string]string)
-		for _, v := range re_local {
-			res, err := getRoleEntityPair(v.(string))
-
-			if err != nil {
-				return fmt.Errorf(
-					"Old state has malformed Role/Entity pair: %v", err)
-			}
-
-			re_local_map[res.Entity] = res.Role
-		}
-
+	// The API offers no way to retrieve predefined ACLs,
+	// and we can't tell which access controls were created
+	// by the predefined roles, so...
+	//
+	// This is, needless to say, a bad state of affairs and
+	// should be fixed.
+	if _, ok := d.GetOk("role_entity"); ok {
```

Comment:
It seems we're no long reading `predefined_acl` at all, is that intentional?

Reply:
Yes, the comment above talks about it briefly. but to expand on it here:

* The resource never actually read `predefined_acl`.
* It _still_ does not read `predefined_acl` because it can't. `predefined_acl` is just a shorthand used when creating or updating a bucket, and isn't actually returned by any API call. It just gets translated into a predefined list of ACLs (which, sadly, aren't predictable--as far as I can tell), and is never seen again. This means `predefined_acl` is probably a bad thing to support as such in terraform, but deprecating it entirely felt out of scope for this PR, especially as I think it would take quite a bit of work to replace it.

Reply:
Thanks for explaining!